src/components/FormSettingsUserData.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect, useRef } from "react";
+import {
+  Form,
+  Modal,
+  ModalBody,
+  ModalTitle,
+  Button,
+  Figure,
+  Row,
+} from "react-bootstrap";
+
+// interface IFormInput {
+//   userName: string;
+//   bio: string;
+//   location: string;
+//   website: string;
+//   dateOfBirth: Date;
+// }
+
+const FormSettingsUserData = () => {
+  const [show, setShow] = useState(false);
+  const handleClose = () => setShow(false);
+
+  //------------------------------ validation ------------------------
+
+  //------------------------------ avatar image ------------------------
+
+  const [fileAvatar, setFileAvatar] = useState<File>();
+  const [previewAvatar, setPreviewAvatar] = useState("");
+  const fileRefAvatar = useRef(null);
+
+  useEffect(() => {
+    let objectUrlAvatar: string;
+    if (fileAvatar) {
+      objectUrlAvatar = URL.createObjectURL(fileAvatar);
+      setPreviewAvatar(objectUrlAvatar);
+    }
+
+    return () => URL.revokeObjectURL(objectUrlAvatar);
+  }, [fileAvatar]);
+
+  //------------------------------ userImage ------------------------
+  const [fileImage, setFileImage] = useState<File>();
+  const [previewImage, setPreviewImage] = useState("");
+  const fileRefImage = useRef(null);
+
+  useEffect(() => {
+    let objectUrlImage: string;
+    if (fileImage) {
+      objectUrlImage = URL.createObjectURL(fileImage);
+      setPreviewImage(objectUrlImage);
+    }
+
+    return () => URL.revokeObjectURL(objectUrlImage);
+  }, [fileImage]);
+
+  const handleAvatarUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.currentTarget && e.currentTarget.files) {
+      setFileAvatar(e.currentTarget.files[0]);
+    }
+  };
+
+  const handleImageUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.currentTarget && e.currentTarget.files) {
+      setFileImage(e.currentTarget.files[0]);
+    }
+  };
+  //-----------------------------------------------------------------------
+
+  return (
+    <>
+      {/*----modal window -------*/}
+      <Modal show={show} onHide={handleClose}>
+        <Modal.Header closeButton>
```

Comment:
You could use ModalUniversal to save some time.

Reply:
is this a prerequisite?